modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationRegistry.java

```diff
@@ -96,8 +97,7 @@ public void setConfiguration( final ApplicationKey key, final Configuration conf
                     callInvalidators( k, ApplicationInvalidationLevel.CACHE );
                 }
                 LOG.debug( "Configure and unlatch app {}", key );
-                v.application.setConfig( configuration );
-                v.unlatch();
+                v.setConfig( configuration );
```

Comment:
Config happens after invalidation. Super-fast frontend request may grab old config for ScriptExecutor

Reply:
It is less critical. And it used to be like this before. Submitted new issue #7994